From 98e4c6834f5678d513d85e73388058224dbfe66e Mon Sep 17 00:00:00 2001
From: Thomas Ackermann <th.acker@arcor.de>
Date: Mon, 1 Oct 2012 20:06:48 +0200
Subject: [PATCH] Create pdf and mobi from all html files

- combine all html files into a single html file; insert page breaks to separate major sections
- use wkhtmltopdf (V0.9.9 "patched QT") to transform html into pdf
- use kindlegen (V2.8) to transform html into mobi
---
 Documentation/.gitignore      |  2 ++
 Documentation/footerend.txt   |  4 +++
 Documentation/footerstart.txt |  7 +++++
 Documentation/makedocpdf.sh   | 68 +++++++++++++++++++++++++++++++++++++++++++
 Documentation/pagebreak.txt   |  1 +
 5 files changed, 82 insertions(+)
 create mode 100644 Documentation/footerend.txt
 create mode 100644 Documentation/footerstart.txt
 create mode 100755 Documentation/makedocpdf.sh
 create mode 100644 Documentation/pagebreak.txt

diff --git a/Documentation/.gitignore b/Documentation/.gitignore
index 2c8b2d6..fb90bc1 100644
--- a/Documentation/.gitignore
+++ b/Documentation/.gitignore
@@ -4,6 +4,7 @@
 *.made
 *.texi
 *.pdf
+*.mobi
 git.info
 gitman.info
 howto-index.txt
@@ -11,3 +12,4 @@ doc.dep
 cmds-*.txt
 mergetools-*.txt
 manpage-base-url.xsl
+docfiles.txt
diff --git a/Documentation/footerend.txt b/Documentation/footerend.txt
new file mode 100644
index 0000000..ed16923
--- /dev/null
+++ b/Documentation/footerend.txt
@@ -0,0 +1,4 @@
+    </td>
+  </tr>
+</table>
+</body></html>
diff --git a/Documentation/footerstart.txt b/Documentation/footerstart.txt
new file mode 100644
index 0000000..a2746ef
--- /dev/null
+++ b/Documentation/footerstart.txt
@@ -0,0 +1,7 @@
+<html>
+<head></head>
+<body style="border:0; margin: 0;" onload="subst()">
+<table style="border-top: 1px solid black; width: 100%">
+  <tr>
+    <td class="section"></td>
+    <td style="text-align:center">
diff --git a/Documentation/makedocpdf.sh b/Documentation/makedocpdf.sh
new file mode 100755
index 0000000..fb8b936
--- /dev/null
+++ b/Documentation/makedocpdf.sh
@@ -0,0 +1,68 @@
+#!/bin/sh
+
+rm -f git-doc.pdf
+rm -f git-doc.html
+rm -f git-docs.html
+rm -f docfiles.txt
+rm -f pagebreak.html
+rm -f footer.html
+
+cat pagebreak.txt                  >pagebreak.html
+
+ls gittutorial.html                >>docfiles.txt
+ls gittutorial-2.html              >>docfiles.txt
+ls everyday.html                   >>docfiles.txt
+ls gitworkflows.html               >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls git.html                        >>docfiles.txt
+ls git-a*.html                     >>docfiles.txt
+ls git-bisect.html                 >>docfiles.txt
+ls git-b[j-z]*.html                >>docfiles.txt
+ls git-[c-s]*.html                 >>docfiles.txt
+ls git-ta*.html                    >>docfiles.txt
+ls gitk.html                       >>docfiles.txt
+ls git-[u-z]*.html                 >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls gita*.html                      >>docfiles.txt
+ls git-bisect-lk2009.html          >>docfiles.txt
+ls git[cdghimnr]*.html             >>docfiles.txt
+ls git-tools.html                  >>docfiles.txt
+ls gitweb*.html                    >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls howto-index.html                >>docfiles.txt
+ls howto/*.html                    >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls technical/api-index.html        >>docfiles.txt
+ls technical/api-[a-h]*.html       >>docfiles.txt
+ls technical/api-in-*.html         >>docfiles.txt
+ls technical/api-[j-z]*.html       >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls technical/[b-z]*.html           >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls install.html                    >>docfiles.txt
+
+ls pagebreak.html                  >>docfiles.txt
+ls RelNotes/*.html                 >>docfiles.txt
+
+cat footerstart.txt                >footer.html
+cat ../GIT-VERSION-FILE            >>footer.html
+cat footerend.txt                  >>footer.html
+
+echo "Creating user-manual.mobi ..."
+kindlegen -c2 user-manual.html -o user-manual.mobi
+
+echo "Creating git-doc.html ..."
+cat docfiles.txt | xargs cat | cat >git-docs.html
+cat readme.html git-docs.html >git-doc.html
+
+echo "Creating git-doc.mobi ..."
+kindlegen -c2 git-doc.html -o git-doc.mobi
+
+echo "Creating git-doc.pdf ..."
+cat docfiles.txt | xargs cat | wkhtmltopdf --toc --cover readme.html --footer-html footer.html --disable-external-links - git-doc.pdf
diff --git a/Documentation/pagebreak.txt b/Documentation/pagebreak.txt
new file mode 100644
index 0000000..fdaffd1
--- /dev/null
+++ b/Documentation/pagebreak.txt
@@ -0,0 +1 @@
+</p style="page-break-after: always">
-- 
1.8.1.msysgit.1

